embassy-nrf/src/uarte.rs

```diff
@@ -52,6 +52,39 @@ impl Default for Config {
     }
 }
 
+bitflags::bitflags! {
+    /// Error source flags
+    pub struct ErrorSource: u32 {
+        /// Buffer overrun
+        const OVERRUN = 0x01;
+        /// Parity error
+        const PARITY = 0x02;
+        /// Framing error
+        const FRAMING = 0x04;
+        /// Break condition
+        const BREAK = 0x08;
+    }
+}
+
+impl TryFrom<ErrorSource> for () {
```

Comment:
this is a bit strange. I'd suggest adding a method `check(self) -> Result<(), Error>` to `ErrorSource` instead.